R/pkg/R/DataFrame.R

```diff
@@ -1217,29 +1217,38 @@ setMethod("[[", signature(x = "DataFrame", i = "numericOrcharacter"),
 
 #' @rdname subset
 #' @name [
-setMethod("[", signature(x = "DataFrame", i = "missing"),
-          function(x, i, j, ...) {
-            if (is.numeric(j)) {
-              cols <- columns(x)
-              j <- cols[j]
-            }
-            if (length(j) > 1) {
-              j <- as.list(j)
+setMethod("[", signature(x = "DataFrame"),
+          function(x, i, j, ..., drop=T) {
+            # Perform filtering first if needed
```

Comment:
Also I think `drop=F` should be default if we want to maintain backwards compatibility.